### ISSUE TITLE
SPIKE: Viability of moving SiS to a microservice

### ISSUE THREAD

**Issue by ianhundere** [department-of-veterans-affairs/va.gov-team#47538]
## Description
This is an in-tandem ticket with [SPIKE: Research SiS Controller](https://app.zenhub.com/workspaces/identity-5f5bab705a94c9001ba33734/issues/department-of-veterans-affairs/va.gov-team/46952) whereby we'll be looking at the viability of transitioning SiS to a microservice hosted in EKS.

The focus of this will be looking at the current architecture of `vets-api` and to understand how we should best approach moving SiS to its own microservice.

## How is the code modularized within `vets-api`?
![image.png](https://images.zenhubusercontent.com/624b40ac4980d90d75917e4c/9346b9d9-a51c-4a43-a91d-5d6be4215f14)

tl;dr
what do we have, what do we need?

## Acceptance Criteria
Understanding the following:
- Current architecture:
    - [ ] controller / service(s) running in a dockerized monolith (using AR models)
- What can we use that's already there:
    - [ ] sidekiq
- Solutions:
    - [ ] packaged Gem
- What are the bottlenecks:
    - [ ] ...
## Resources
- [How to Break Apart a Rails Monolith
](https://kellysutton.com/2020/03/12/how-to-break-apart-a-rails-monolith.html)

**Comment by ianhundere** [comment]
closing this and will continue putting notes via https://app.zenhub.com/workspaces/identity-5f5bab705a94c9001ba33734/issues/department-of-veterans-affairs/va.gov-team/46952
